File: ios/Reguerta/Reguerta/Domain/Shifts/ShiftAssignment.swift
import Foundation

enum ShiftType: String, Equatable, Sendable {
    case delivery
    case market
}

enum ShiftStatus: String, Equatable, Sendable {
    case planned
    case swapPending = "swap_pending"
    case confirmed
}

struct ShiftAssignment: Identifiable, Equatable, Sendable {
    let id: String
    let type: ShiftType
    let dateMillis: Int64
    let assignedUserIds: [String]
    let helperUserId: String?
    let status: ShiftStatus
    let source: String
    let createdAtMillis: Int64
    let updatedAtMillis: Int64

    func isAssigned(to userId: String) -> Bool {
        assignedUserIds.contains(userId) || helperUserId == userId
    }
}

File: android/Reguerta/app/src/main/java/com/reguerta/user/domain/shifts/ShiftAssignment.kt
package com.reguerta.user.domain.shifts

enum class ShiftType {
    DELIVERY,
    MARKET,
}

enum class ShiftStatus {
    PLANNED,
    SWAP_PENDING,
    CONFIRMED,
}

data class ShiftAssignment(
    val id: String,
    val type: ShiftType,
    val dateMillis: Long,
    val assignedUserIds: List<String>,
    val helperUserId: String?,
    val status: ShiftStatus,
    val source: String,
    val createdAtMillis: Long,
    val updatedAtMillis: Long,
) {
    fun isAssignedTo(userId: String): Boolean =
        assignedUserIds.contains(userId) || helperUserId == userId
}
